### ISSUE TITLE
Gain keeps getting higher every time a AudioMotionAnalyzer is initialized

### ISSUE THREAD

**Issue by lucienimmink** [hvianna/audioMotion-analyzer#53]
Hi,

First of all: great software! I'm implementing this with Lit with a web component that is only added to the DOM if needed.
I have some hacky work-around to bypass having to disconnect on destroy of the component

```javascript
let audioCtx;
let source;
if (!window._audioCtx) {
          audioCtx = new AudioContext();
          source = audioCtx.createMediaElementSource(window._player);

          // store in memory for reuse
          window._audioCtx = audioCtx;
          window._source = source;
        } else {
          audioCtx = window._audioCtx;
          source = window._source;
        }
```

Every time I call `new AudioMotionAnalyzer()` the gain of this chain is increased and it starts clipping. Setting the volume has no effect.

```javascript
const visualizer = new AudioMotionAnalyzer(canvas, {
          audioCtx,
          source,
});
```

**Comment by hvianna** [comment]
The analyzer is connected to the speakers by default, so multiple calls to the constructor will multiply the output as well. Also, depending on the player you're using, it may already be doing the connection to the *AudioContext* **destination** node (speakers).

You can use [`connectSpeakers: false`](https://audiomotion.dev/#/?id=connectspeakers-boolean) in the constructor options to skip connecting the analyzer to the speakers, like so:

```js
const visualizer = new AudioMotionAnalyzer(canvas, {
          audioCtx,
          source,
          connectSpeakers: false
});
```

If you get no sound after that, connect your source node to the destination, like so:

```js
source.connect( audioCtx.destination );
```

Do you really need to call the constructor multiple times, though? Like, you need multiple visualizations running at the same time? Please note these instances will keep consuming resources, unless you use [`toggleAnalyzer()`](https://audiomotion.dev/#/?id=toggleanalyzer-boolean-) to stop the audio processing.

Anyway, a `destroy()` method is a good suggestion! I'll look into adding this in a future release! 👍

**Comment by lucienimmink** [comment]
I'll look into your thought, for now just want to come back to the remark if I need more visualisations at the same time: no :)
I have a music player with an optional visualiser. If shown it's added to the Dom as a web component and if removed, the component is removed from the dom, so I can call destroy() when the web component is removed, all the while the music keeps on playing.

**Comment by lucienimmink** [comment]
Just FYI:
Using the `connectSpeaker: false` option is (for me) fixing the issue, although I believe it would be a better/safer option to have the proper `destroy()` method.
Thank you for the tip!

**Comment by hvianna** [comment]
Cool, thanks for the follow-up!

**Comment by hvianna** [comment]
`destroy()` is now available in [v4.2.0](https://github.com/hvianna/audioMotion-analyzer/releases/tag/4.2.0) 🎉
